tasty-query/shared/src/main/scala/tastyquery/Symbols.scala

```diff
@@ -208,7 +210,74 @@ object Symbols {
     override def toString: String = "NoSymbol"
   end NoSymbol
 
-  final class TermSymbol private (val name: TermName, override val owner: Symbol) extends Symbol(owner):
+  sealed abstract class TermOrTypeSymbol(override val owner: Symbol) extends Symbol(owner):
+    // Overriding relationships
+
+    /** The non-private symbol whose name and type matches the type of this symbol in the given class.
+      *
+      * @param inClass
+      *   The class containing the result symbol's definition
+      * @param siteClass
+      *   The base class from which member types are computed
+      */
+    private final def matchingDecl(inClass: ClassSymbol, siteClass: ClassSymbol)(using Context): Option[Symbol] =
+      this match
+        case self: TypeSymbol =>
+          inClass.getDecl(self.name).filterNot(_.is(Private))
+        case self: TermSymbol =>
+          val candidates = inClass.getDecls(self.name).asInstanceOf[List[TermSymbol]].filterNot(_.is(Private))
+          if candidates.isEmpty then None
+          else
+            val site = siteClass.thisType
+            val targetType = self.declaredTypeAsSeenFrom(site)
+            candidates.find { candidate =>
+              // TODO Also check targetName here
+              candidate.declaredTypeAsSeenFrom(site).matchesLoosely(targetType)
+            }
+    end matchingDecl
+
+    /** If false, this symbol cannot possibly participate in an override, either as overrider or overridee. */
+    final def canMatchInheritedSymbols(using Context): Boolean =
+      owner.isClass && memberCanMatchInheritedSymbols
+
+    /** If false, this class member cannot possibly participate in an override, either as overrider or overridee. */
+    private final def memberCanMatchInheritedSymbols(using Context): Boolean =
+      !isConstructor && !is(Private)
+
+    private final def isConstructor(using Context): Boolean =
+      name == nme.Constructor
+
+    /** The symbol, in class `inClass`, that is overridden by this symbol, if any. */
+    final def overriddenSymbol(inClass: ClassSymbol)(using Context): Option[Symbol] =
+      if canMatchInheritedSymbols then
+        val ownerClass = owner.asClass
+        if ownerClass.isSubclass(inClass) then matchingDecl(inClass, siteClass = ownerClass)
+        else None
+      else None
```

Comment:
This seems to match dotty more - so I guess it will match constructors when inClass is the owner
```suggestion
      else if owner.asClass == inClass then matchingDecl(inClass, siteClass = inClass)
      else None
```

Reply:
I changed it and also added tests for that.